<?xml version="1.0" ?>
<robot name="ur5e">

  <link name="base_link">
    <visual>
      <geometry>
        <cylinder length="0.05" radius="0.06"/>
      </geometry>
    </visual>
  </link>

  <joint name="shoulder_pan_joint" type="revolute">
    <parent link="base_link"/>
    <child link="shoulder_link"/>
    <origin rpy="0.0 0.0 0.0" xyz="0.0 0.0 0.1625"/>
    <axis xyz="0 0 1"/>
    <limit effort="150.0" lower="-6.28" upper="6.28" velocity="3.15"/>
  </joint>

  <link name="shoulder_link">
    <visual>
      <geometry>
        <cylinder length="0.15" radius="0.06"/>
      </geometry>
    </visual>
  </link>

  <joint name="shoulder_lift_joint" type="revolute">
    <parent link="shoulder_link"/>
    <child link="upper_arm_link"/>
    <origin rpy="0.0 1.570796325 0.0" xyz="0.0 0.138 0.0"/>
    <axis xyz="0 1 0"/>
    <limit effort="150.0" lower="-6.28" upper="6.28" velocity="3.15"/>
  </joint>

  <link name="upper_arm_link">
    <visual>
      <geometry>
        <cylinder length="0.425" radius="0.05"/>
      </geometry>
    </visual>
  </link>

  <joint name="elbow_joint" type="revolute">
    <parent link="upper_arm_link"/>
    <child link="forearm_link"/>
    <origin rpy="0.0 0.0 0.0" xyz="0.0 -0.131 0.425"/>
    <axis xyz="0 1 0"/>
    <limit effort="150.0" lower="-3.14" upper="3.14" velocity="3.15"/>
  </joint>

  <link name="forearm_link">
    <visual>
      <geometry>
        <cylinder length="0.392" radius="0.04"/>
      </geometry>
    </visual>
  </link>

  <joint name="wrist_1_joint" type="revolute">
    <parent link="forearm_link"/>
    <child link="wrist_1_link"/>
    <origin rpy="0.0 1.570796325 0.0" xyz="0.0 0.0 0.392"/>
    <axis xyz="0 1 0"/>
    <limit effort="28.0" lower="-6.28" upper="6.28" velocity="3.2"/>
  </joint>

  <link name="wrist_1_link">
    <visual>
      <geometry>
        <cylinder length="0.1" radius="0.03"/>
      </geometry>
    </visual>
  </link>

  <joint name="wrist_2_joint" type="revolute">
    <parent link="wrist_1_link"/>
    <child link="wrist_2_link"/>
    <origin rpy="0.0 0.0 0.0" xyz="0.0 0.127 0.0"/>
    <axis xyz="0 0 1"/>
    <limit effort="28.0" lower="-6.28" upper="6.28" velocity="3.2"/>
  </joint>

  <link name="wrist_2_link">
    <visual>
      <geometry>
        <cylinder length="0.1" radius="0.03"/>
      </geometry>
    </visual>
  </link>

  <joint name="wrist_3_joint" type="revolute">
    <parent link="wrist_2_link"/>
    <child link="wrist_3_link"/>
    <origin rpy="0.0 0.0 0.0" xyz="0.0 0.0 0.1"/>
    <axis xyz="0 1 0"/>
    <limit effort="28.0" lower="-6.28" upper="6.28" velocity="3.2"/>
  </joint>

  <link name="wrist_3_link">
     <visual>
      <geometry>
        <box size="0.05 0.05 0.05"/>
      </geometry>
    </visual>
  </link>

</robot>